File: code/frontend/app/src/services/user-stats.ts
import type { StudyCalendarRes } from '@/types/user-stats';

import { apiGet } from './http';

export function getStudyCalendarApi(userId: string, days = 365): Promise<StudyCalendarRes> {
  const q = new URLSearchParams({ days: String(days) });
  return apiGet<StudyCalendarRes>(`/users/${encodeURIComponent(userId)}/study-calendar?${q.toString()}`);
}

File: code/backend/app/api/routes/user_stats.py
from fastapi import APIRouter, Query

from app.services.user_stats_service import get_user_study_calendar
from app.utils.response import success


router = APIRouter(tags=["user-stats"])


@router.get("/users/{user_id}/study-calendar")
def user_study_calendar_api(user_id: str, days: int = Query(default=365, ge=30, le=730)):
    data = get_user_study_calendar(user_id=user_id, days=days)
    return success(data=data)

